tools/AduCmdlets/AduAzStorageBlobHelper.psm1

```diff
@@ -0,0 +1,80 @@
+#
+# Azure Device Update for IoT Hub
+# Powershell module for uploading file to Azure Storage Blob Container.
+# Copyright (c) Microsoft Corporation.
+#
+
+#Requires -Version 5.0
+#Requires -Modules Az.Accounts, Az.Storage
+
+function Get-AduAzBlobContainer
+{
+<#
+    .SYNOPSIS
+        Get a reference to Azure Storage Blob container object.
+#>
+    [CmdletBinding()]
+    Param(
+        # Azure subscription id.
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        $SubscriptionId,
+
+        # Azure resource group name of the Azure Storage account.
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        $ResourceGroupName,
+
+        # Azure Storage account name.
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        $StorageAccountName,
+
+        # Blob container name.
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        $ContainerName,
+
+        # Skip connecting to Azure.
+        [switch] $SkipLogin
+    )
+
+    if (!$SkipLogin)
+    {
+        Connect-AzAccount -ErrorAction Stop -UseDeviceAuthentication | Out-Null
+        $subscription = Get-AzSubscription -SubscriptionId $SubscriptionId -ErrorAction Stop
+        Set-AzContext $subscription -ErrorAction Stop | Out-Null
+    }
+
+    $account = Get-AzStorageAccount -ResourceGroupName $ResourceGroupName -Name $StorageAccountName -ErrorAction Stop
+    $container = New-AzStorageContainer -Name $ContainerName -Context $account.Context -ErrorAction SilentlyContinue
```

Comment:
>-ErrorAction SilentlyContinue [](http://example.com/codeflow?start=87&length=29)

why silently continue? I realize that 51 will ultimately fail if this does, but you might mask the root cause of failure, no?

Reply:
new-azstoragecontainer throws error when container already exists.